recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,1418 @@
+    "tiff",
+    "utf8",
+    "UtilitiesBenchmarks",
+    "verdict",
+    "ViewsContext2D",
+    "ViewsCore",
+    "ViewsInfovis",
+    "ViewsQt",
+    "vpic",
+    "vtkm",
+    "vtksys",
+    "WebCore",
+    "WebGLExporter",
+    "WebPython",
+    "WrappingPythonCore",
+    "WrappingTools",
+    "xdmf2",
+    "xdmf3",
+    "zfp",
+    "zlib",
+]
+
+
+
+class VtkConan(ConanFile):
+    name = "vtk"
+    description = "The Visualization Toolkit (VTK) by Kitware is an open-source, \
+        freely available software system for 3D computer graphics, \
+        image processing, and visualization."
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.vtk.org/"
+    topics = ("scientific", "image", "processing", "visualization")
+    settings = "os", "compiler", "build_type", "arch"
+
+    short_paths = True
+
+    # Alternative method: can use git directly - helpful when hacking VTK
+    # TODO allow user to set the git_url from the command line, during conan install
+    # git_url = "https://gitlab.kitware.com/vtk/vtk.git"
+    git_url = "/build/git-mirrors/vtk.git"
+
+
+    ############
+    # list of all possible 3rd parties (as of 9.1.0)
+    # with each one marked as where they should come from.
+    #
+    # Procedure for this list:
+    #  - for each item, check if there is a recipe in CCI (clone it for easy search)
+    #  - check VTK/ThirdParty/PACKAGE/CMakeLists.txt for minimum version
+    #  - check VTK/ThirdParty/PACKAGE/vtkPACK/README.kitware.md
+    #       for any notes on special changes to the library,
+    #       apart from integration with the build system.
+    #  - try adding to the "parties" list
+    #  - ensure VTK_MODULE_USE_EXTERNAL_VTK_pack is NOT set to False
+    #  - 'conan create' with the group/module enabled that requires this pack
+    #  - build with CMAKE_FIND_DEBUG_MODE=True (option below to uncomment)
+    #       and carefully ensure it isn't picking up any system libs
+    #       (search the output for "The item was")
+    #
+    # cgns:             conan (cgns)
+    # cli11:            conan (cli11)
+    # diy2:             internal (not available in CCI - TODO)
+    # doubleconversion: conan (double-conversion)
+    # eigen:            conan (eigen)
+    # exodusII:         internal (not available in CCI - TODO)
+    # expat:            conan (expat)
+    # exprtk:           conan (exprtk)
+    # fides:            internal (not available in CCI - TODO)
+    # fmt:              conan (fmt)
+    # freetype:         conan (freetype)
+    # gl2ps:            internal (not available in CCI - TODO)
+    # glew:             conan (glew)
+    # h5part:           internal (not available in CCI - TODO)
+    # hdf5:             conan (hdf5)
+    # ioss:             internal (not available in CCI - TODO)
+    # jpeg:             conan (libjpeg or libjpeg-turbo)
+    # jsoncpp:          conan (jsoncpp)
+    # kissfft:          conan (kissfft)
```

Comment:
vtk doesn't look for external kissfft and always uses internal one

Reply:
Seems to be working with 2 additional changes:

```
--- a/ThirdParty/kissfft/CMakeLists.txt
+++ b/ThirdParty/kissfft/CMakeLists.txt
@@ -1,9 +1,15 @@
 set(KISSFFT_SCALAR_TYPE "double")
 
-vtk_module_third_party_internal(
-  LICENSE_FILES "vtkkissfft/COPYING"
-  VERSION       "master"
-  STANDARD_INCLUDE_DIRS)
+vtk_module_third_party(
+  INTERNAL
+    LICENSE_FILES "vtkkissfft/COPYING"
+    VERSION       "master"
+    STANDARD_INCLUDE_DIRS
+  EXTERNAL
+    PACKAGE kissfft
+    TARGETS kissfft::kissfft-double
+            kissfft-double
+    STANDARD_INCLUDE_DIRS)
 
 configure_file(
   "${CMAKE_CURRENT_SOURCE_DIR}/vtk_kissfft.h.in"

```
```
--- a/Common/Math/vtkFFT.h
+++ b/Common/Math/vtkFFT.h
@@ -36,7 +36,7 @@
 #include "vtk_kissfft.h" // For kiss_fft_scalar, kiss_fft_cpx
 // clang-format off
 #include VTK_KISSFFT_HEADER(kiss_fft.h)
-#include VTK_KISSFFT_HEADER(tools/kiss_fftr.h)
+#include VTK_KISSFFT_HEADER(kiss_fftr.h)
 // clang-format on
 
 #include <vector> // For std::vector

```

Reply:
Really?  That's weird, I thought it was picking up the conan kissfft.

Reply:
Thanks again, finally incorporated and started to test.